src/inbox/mod_inbox.erl

```diff
@@ -232,50 +229,67 @@ send_message(Acc, To = #jid{lserver = LServer}, Msg) ->
 
 %%%%%%%%%%%%%%%%%%%
 %% Handlers
--spec user_send_packet(Acc :: mongoose_acc:t(), From :: jid:jid(),
-                       To :: jid:jid(), Packet :: exml:element()) ->
-    mongoose_acc:t().
-user_send_packet(Acc, From, To, #xmlel{name = <<"message">>} = Msg) ->
-    maybe_process_message(Acc, From, To, Msg, outgoing);
-user_send_packet(Acc, _From, _To, _Packet) ->
-    Acc.
-
--spec inbox_unread_count(Acc :: mongoose_acc:t(), To :: jid:jid()) -> mongoose_acc:t().
-inbox_unread_count(Acc, To) ->
+-spec user_send_packet(Acc, Args, Extra) -> {ok, Acc} when
+      Acc :: mongoose_acc:t(),
+      Args :: #{from := jid:jid(), to := jid:jid(), packet := exml:element()},
+      Extra :: map().
+user_send_packet(Acc, #{from := From, to := To, packet := #xmlel{name = <<"message">>} = Msg}, _) ->
+    NewAcc = maybe_process_message(Acc, From, To, Msg, outgoing),
+    {ok, NewAcc};
+user_send_packet(Acc, _, _) ->
+    {ok, Acc}.
+
+-spec inbox_unread_count(Acc, Params, Extra) -> {ok, Acc} when
+      Acc :: mongoose_acc:t(),
+      Params :: #{user := jid:jid()},
+      Extra :: map().
+inbox_unread_count(Acc, #{user := User}, _) ->
     Res = mongoose_acc:get(inbox, unread_count, undefined, Acc),
-    get_inbox_unread(Res, Acc, To).
-
--spec filter_local_packet(mongoose_hooks:filter_packet_acc() | drop) ->
-    mongoose_hooks:filter_packet_acc() | drop.
-filter_local_packet(drop) ->
-    drop;
-filter_local_packet({From, To, Acc, Msg = #xmlel{name = <<"message">>}}) ->
+    NewAcc = get_inbox_unread(Res, Acc, User),
+    {ok, NewAcc}.
+
+-spec filter_local_packet(drop, _, _) -> {ok, drop};
```

Comment:
Actually as a simplification for this, it could all be in two clauses, like
```erl
filter_local_packet({From, To, Acc, Msg = #xmlel{name = <<"message">>}}, _, _) ->
    ...
filter_local_packet(Filter, _, _) ->
    {ok, Filter}.
```

And that second clause will cover both the `drop` key, and the filter when the packet is not a message. Also less matching for the runtime, save a CPU cycle or two 😄

Reply:
fixed :) we can merge it now